src/store/index.ts

```diff
@@ -0,0 +1,344 @@
+      state.usedConfig = null
+      state.connection = null
+      state.database = null
+    },
+    updateConnection(state, {connection, database}) {
+      state.connection = connection
+      state.database = database
+    },
+    tables(state, tables) {
+      state.tables = tables
+    },
+    tablesLoading(state, value: string) {
+      state.tablesLoading = value
+    },
+    addPinned(state, table: string) {
+      if (state.database && state.pinStore[state.database] === null) {
+        Vue.set(state.pinStore, state.database, [])
+      }
+      if (state.database && !state.pinStore[state.database].includes(table)) {
+        state.pinStore[state.database].push(table)
+      }
+    },
+    removePinned(state, table) {
+      if (state.database === null || state.pinStore[state.database] === null) {
+        return
+      }
+      Vue.set(state.pinStore, state.database, _.without(state.pinStore[state.database], table))
+    },
+    config(state, newConfig) {
+      if (!state.connectionConfigs.includes(newConfig)) {
+        state.connectionConfigs.push(newConfig)
+      }
+    },
+    removeConfig(state, config) {
+      state.connectionConfigs = _.without(state.connectionConfigs, config)
+    },
+    removeUsedConfig(state, config) {
+      state.usedConfigs = _.without(state.usedConfigs, config)
+    },
+    configs(state, configs: UsedConnection[]){
+      Vue.set(state, 'connectionConfigs', configs)
+    },
+    usedConfigs(state, configs: UsedConnection[]) {
+      Vue.set(state, 'usedConfigs', configs)
+    },
+    history(state: State, history) {
+      state.history = history
+    },
+    historyAdd(state: State, run: UsedQuery) {
+      state.history.unshift(run)
+    },
+    historyRemove(state, historyQuery) {
+      state.history = _.without(state.history, historyQuery)
+    },
+    favorites(state: State, list) {
+      state.favorites = list
+    },
+    favoritesAdd(state: State, query) {
+      state.favorites.unshift(query)
+    },
+    removeUsedFavorite(state: State, favorite) {
+      state.favorites = _.without(state.favorites, favorite)
+    },
+
+  },
+  actions: {
+
+    async test(context, config: SavedConnection) {
+      // TODO (matthew): fix this mess.
+      if (context.state.username && config.defaultDatabase) {
+        const server = ConnectionProvider.for(config, context.state.username)
+        await server?.createConnection(config.defaultDatabase).connect()
+        server.disconnect()
+      }
+    },
+
+    async fetchUsername(context) {
+      const name = await username()
+      context.commit('setUsername', name)
+    },
+
+    async connect(context, config: SavedConnection) {
+      if (context.state.username) {
+        const server = ConnectionProvider.for(config, context.state.username)
+        // TODO: (geovannimp) Check case connection is been created with undefined as key
+        const connection = server.createConnection(config.defaultDatabase || undefined)
```

Comment:
I think this is ok?

Reply:
The problem is this:
```
if (server.db[dbName]) {
```

As dbName can be `undefined`, this can happen: `server.db[undefined]`.

It's actually working I think, but typescript complains about using `undefined` as key.

```
Type 'undefined' cannot be used as an index type.ts(2538)
```

We can keep this way for now, but would be good to refactor any time.